grails-app/services/org/pih/warehouse/core/LocationService.groovy

```diff
@@ -167,7 +171,7 @@ class LocationService {
         def nullHigh = new NullComparator(true)
         def locations = getLoginLocations(currentLocation)
         if (locations) {
-			locations = locations.findAll { Location location -> user.getEffectiveRoles(location) }
+			locations = locations.findAll { Location location -> user.getEffectiveRoles(location) && user.hasPrimaryRole(location) }
```

Comment:
We actually just need `user.hasPrimaryRole(location)` since the method calls getEffectiveRoles. In other words, hasPrimaryRole will return false in exactly the same situation that getEffectiveRole returns false (i.e. when effective roles has 0 items).